packages/playground/src/utils/nodeSelector.ts

```diff
@@ -199,6 +199,30 @@ export function loadNodes(gridStore: ReturnType<typeof useGrid>, filters: Filter
   return gridStore.client.capacity.filterNodes(filters);
 }
 
+export async function validateRentContract(
+  gridStore: ReturnType<typeof useGrid>,
+  node: NodeInfo | undefined | null,
+): Promise<true> | never {
+  if (!node || !node.nodeId) {
+    throw "Node ID is required.";
+  }
+
+  try {
+    const contractInfo = await gridStore.grid.contracts.get({ id: node.rentContractId });
+    if (contractInfo.state.gracePeriod) {
```

Comment:
This makes all deployments fail. 

Reply:
We can skip this check if the rent contract is 0 as the node in this case is not rented.

Reply:
Its should be fixed in https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2431/files#diff-c5cb2a07f18e9933e21237d4b7caeac5ea18ab5a451b4cd1d5cf73472dffaa4fL83-R83